src/python/pants/jvm/jdk_rules.py

```diff
@@ -87,10 +100,34 @@ async def setup_jdk(coursier: Coursier, jvm: JvmSubsystem, bash: BashBinary) ->
         ),
     )
 
+    return Nailgun(nailgun)
+
+
+@rule
+async def global_jdk(jvm: JvmSubsystem) -> JdkRequest:
+    """Creates a `JdkRequest` object based on the JVM subsystem options.
+
+    This is effectively a singleton for now, but by the time we complete multiple JVM support, it
+    won't be.
+    """
+
     if jvm.jdk == "system":
+        return JdkRequest(major_version="", is_system=True)
+    else:
+        return JdkRequest(major_version=jvm.jdk, is_system=False)
```

Comment:
Nit, and fine to ignore, but FYI style guide says to prefer conditional expressions: https://www.pantsbuild.org/docs/style-guide#prefer-conditional-expressions-ternary-expressions

Reply:
Ignoring, this function will disappear in future follow-up PRs.